### PR TITLE
Cars and can

### PR DESCRIPTION
**Pull request by OrangeTux**

### CODE REVIEW

File: content/post/cars-and-can.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+In [previous post]({{< relref "post/obd_ii_elm327_and_python.md">}}) of this
+series I demonstrated how to obtain live data of a car by using an OBD-II
+adapter. I was a bit disappointend about the fact that I could only read a
+handfull of parameters. Together with Wisse Hooghiem I went a step further by
+hooking up in the CAN network of my car. In this post I'll demonstrate how to
+interact with a car's CAN bus in Linux with help of a an Arduino Uno.
+
+## CAN bus
+
+Every car contains electronic chips to control parts of the car. The are called
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: As a narrative note, I should start this paragraph with `First, what is CAN?` or something like that. Now, you're starting right away and that does read somewhat strange.

File: content/post/cars-and-can.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+description = "Use a cars CAN bus to read live data."
+title = "TED2: Cars and CAN"
+series = ["Ted"]
+slug = "cars-and-can"
+author = "Auke Willem Oosterhoff"
++++
+
+In [previous post]({{< relref "post/obd_ii_elm327_and_python.md">}}) of this
+series I demonstrated how to obtain live data of a car by using an OBD-II
+adapter. I was a bit disappointend about the fact that I could only read a
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: `disappointend` -> `disappointed`

File: content/post/cars-and-can.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+handfull of parameters. Together with Wisse Hooghiem I went a step further by
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: - `handfull -> `handful`
-  It's nice to refer to Wisse using his Github profile, so people are getting a slightly better idea who he is :p.

File: content/post/cars-and-can.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+* Motor Control Unit (MCU) - responsible for things like fuel injection and
+  engine speed monitoring
+* Door Control Unit (DCU) - handles the (un)locking of the doors and the
+  windows
+* Tranmission Control Unit (TCU) - controlls the operations related to the tranmsission
+
+Most ECUs require input from other ECUs to operate. For example the Airbag
+Control Unit requires information about speed, acceleration and occupant
+status. Therefore in most cars the ECUs are connected using a CAN bus.
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: Add a comma after Therefore

File: content/post/cars-and-can.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+CAN is an acronym for Controller Area Network and it has beend developed by the
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: `beend` -> `been`

File: content/post/cars-and-can.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+German company Bosch. CAN is a simple and cheap 2-wire bus. Its first use in
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: What does `cheap` mean in this context? Car manafucters don't have pay tons of money to implement it in their cars ;)? In other words, what's the value of mentioning it?

File: content/post/cars-and-can.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+cars dates back to 1990.
+
+## Data frames
+
+CAN has 4 packet types. But in this post we're only interested in 1 type: the
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: Why are you interested in 1 packet type? What are packages types and why aren't you interested in them? I think the reader would be interested to know.

**Review by Appeltabak — CHANGES_REQUESTED** [review]

File: content/post/cars-and-can.md (hunk 3 above)
Comment (OrangeTux) [reply content/post/cars-and-can.md]: Well, I asked him this. But he told me that his name was enough.

File: content/post/cars-and-can.md (hunk 7 above)
Comment (OrangeTux) [reply content/post/cars-and-can.md]: Good point, I left the word `cheap` out.

File: content/post/cars-and-can.md (hunk 6 above)
Comment (OrangeTux) [reply content/post/cars-and-can.md]: Good point, I left the word cheap out.

File: content/post/cars-and-can.md (hunk 7 above)
Comment (OrangeTux) [reply content/post/cars-and-can.md]: Good point. I added a few lines.

File: content/post/cars-and-can.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+* Data Length
+* Data
+
+All parts are marked in the image below. This image was taken from
+[Wikipedia][wikipedia].
+
+{{<figure src="/img/can_data_frame_layout.png">}}
+
+The arbitration id defines the meaning of the data. For example in my car the
+data frames with id 0x2F1 contain the steering angle. Normally this id is 11
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: ` is 11 bits in length` - > `has a length of 11 bits`

Comment (Appeltabak) [inline content/post/cars-and-can.md]: `Normally` -> `Generally`, als je 'In het algemeen...' bedoelt

File: content/post/cars-and-can.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+This intro into CAN is fairly brief. If you want to learn more about this topic
+I recommend reading [Wikipedia page about CAN][wikipedia] or [chapter
+2][bus-protocols] of [The Car Hacker's Handbook][car-hackers-handbook] by Craig
+Smith.
+
+## MCP2515
+
+Your Linux machine probably doesn't have a CAN bus. In order to receive the
+data frames on your machine additional hardware is required. I used a [CAN bus
+shield][can-bus-shield] mounted on top of an Arduino Uno.
```
Comment (Appeltabak) [inline content/post/cars-and-can.md:77]: Foto van deze opstelling?

File: content/post/cars-and-can.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+The shield must be connected to the CAN bus. If your car has a CAN bus it's
+very likely that the bus is wired to the OBD-II connector. Pins 6 and 14 are
+reserved for the CAN bus as shown in the image which is from [The Car Hacker's
+Handbook][car-hackers-handbook].
+
+{{<figure src="/img/obd-ii-pin-out.jpg">}}
+
+The main component of the shield is the [MCP2515][mcp2515]. This is a
+CAN-controller with a SPI interface. I haven't tried, but I guess any other
```
Comment (Appeltabak) [inline content/post/cars-and-can.md]: Remove `-`. that's a Dutch thing.

**Review by Appeltabak — CHANGES_REQUESTED** [review]

**Pull request merged**